packages/lesswrong/components/users/CrosspostLoginPage.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState } from "react";
+import { Components, registerComponent } from "../../lib/vulcan-lib";
+import Button from "@material-ui/core/Button";
+import { useCurrentUser } from "../common/withUser";
+import { useGetParameter } from "../hooks/useGetParameter";
+import { forumHeaderTitleSetting } from "../common/Header";
+import { forumTypeSetting } from "../../lib/instanceSettings";
+import { gql, useMutation } from "@apollo/client";
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: {
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "center",
+  },
+  heading: {
+    display: "flex",
+    flexDirection: "row",
+    alignItems: "center",
+    marginTop: "1.5em",
+  },
+  headingText: {
+    fontSize: 20,
+    margin: 0,
+  },
+  buttonContainer: {
+    marginTop: "1.5em",
+  },
+  error: {
+    color: theme.palette.error.main,
+    marginTop: 16,
+  },
+});
+
+const connectCrossposterMutation = gql`
+  mutation connectCrossposter($token: String) {
+    connectCrossposter(token: $token)
+  }
+`;
+
+const CrosspostLoginPage = ({classes}: {
+  classes: ClassesType,
+}) => {
+  const [mutate, loading] = useMutation(connectCrossposterMutation, {errorPolicy: "all"});
```

Comment:
nit: would prefer `mutate` to be called something descriptive (i.e. `connectCrossposter`)